File: sdk-node/examples/health-check.ts
/**
 * Health check example — demonstrates health() for full gateway status
 * and isHealthy() for a quick boolean check.
 */

import { GatewayClient } from '@c6web/guardrails';

const client = new GatewayClient();

// Full health status
const status = await client.health();
console.log('Status:', status.status);
console.log('Data DB:', status.data_db);
console.log('Log DB:', status.log_db);
console.log('Detection degraded:', status.detection_degraded);

// Quick boolean check
const healthy = await client.isHealthy();
console.log('Is healthy:', healthy);

File: sdk-python/examples/health_check.py
"""Demonstrates checking the C6 Guardrails health status.

Shows:
- Retrieving full health status via client.health().
- Quick boolean check via client.is_healthy().
- Printing detailed health fields.
- Using asyncio.run().
"""

import asyncio

from c6_guardrails import GatewayClient, GatewayClientOptions


async def main() -> None:
    client = GatewayClient(GatewayClientOptions(base_url="http://localhost:8082", api_key="ak_..."))

    # Full health status
    health = await client.health()
    print(f"Status: {health.status}")
    print(f"Data DB: {health.data_db}")
    print(f"Log DB: {health.log_db}")
    print(f"Detection degraded: {health.detection_degraded}")

    # Quick boolean check
    ok = await client.is_healthy()
    print(f"Is healthy: {ok}")

    await client.close()


asyncio.run(main())
